twenty-first/src/math/digest.rs

```diff
@@ -416,4 +472,205 @@ pub(crate) mod digest_tests {
         let bytes_again: [u8; Digest::BYTES] = digest.into();
         prop_assert_eq!(bytes, bytes_again);
     }
+
+    // note: this test panics due to:
+    //       https://github.com/Neptune-Crypto/twenty-first/issues/195
+    //
+    //       for now we use a #[should_panic]. Depending on resolution
+    //       of #195, we may be able to remove it.
+    #[test]
+    #[should_panic]
+    pub fn bytes_in_matches_bytes_out() {
+        let bytes1: [u8; 40] = [255; 40];
+        let d1 = Digest::from(bytes1);
+
+        let bytes2: [u8; 40] = d1.into();
+        let d2 = Digest::from(bytes2);
+
+        println!("bytes1: {:?}", bytes1);
+        println!("bytes2: {:?}", bytes2);
+
+        assert_eq!(d1, d2);
+        assert_eq!(bytes1, bytes2);
+    }
+
+    // note: this test panics due to:
+    //       https://github.com/Neptune-Crypto/twenty-first/issues/195
+    //
+    //       for now we use a #[should_panic]. Depending on resolution
+    //       of #195, we may be able to remove it.
+    #[test]
+    #[should_panic]
+    pub fn bytes_in_matches_bytes_out_extended() {
+        for x in 0..40 {
+            let mut bytes = vec![];
+            for y in 0..40 {
+                match y < x {
+                    true => bytes.push(255),
+                    false => bytes.push(254),
+                }
+            }
+            let bytes1 = <[u8; 40]>::try_from(bytes).unwrap();
+
+            let d1 = Digest::from(bytes1);
+            let bytes2: [u8; 40] = d1.into();
+
+            let d2 = Digest::from(bytes2);
+
+            if bytes1 != bytes2 {
+                println!("bytes1: {:?}", bytes1);
+                println!("bytes2: {:?}", bytes2);
+            }
+
+            assert_eq!(d1, d2);
+            assert_eq!(bytes1, bytes2);
+        }
+    }
+
+    mod hex_test {
+
+        use super::*;
+
+        pub(super) fn hex_examples() -> Vec<(Digest, &'static str)> {
+            vec![
+                (
+                    Digest::default(),
+                    "00000000000000000000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
nit: This is longer than the maximum line width, which is 100 characters. Perhaps a different arrangement reduces the indentation level?

Reply:
this has been through `cargo fmt`.  I'm open to suggestions.

possibly a const vec instead of the hex_examples() fn.   I'll play with it.

Reply:
I addressed this with:

```
                    concat!(
                        "0000000000000000000000000000000000000000",
                        "0000000000000000000000000000000000000000"
                    ),
```

Reply:
Even though it requires naming more things, I find the following arrangement visually pleasing, too. But we're definitely in the “kinda whatever :shrug:” area now. :relieved: 

```rust
pub(super) fn hex_examples() -> Vec<(Digest, &'static str)> {
    let example_0 = (
        Digest::default(),
        "0000000000000000000000000000000000000000\
         0000000000000000000000000000000000000000",
    );
    let example_1 = /* … */;

    // note: this would cause test failure due to:
    //       https://github.com/Neptune-Crypto/twenty-first/issues/195
    let _example_2 = (
        Digest::new(bfe_array![0, 1, 10, 15, 255]),
        "ffffffffffffffffffffffffffffffffffffffff\
         ffffffffffffffffffffffffffffffffffffffff",
    );

    vec![example_0, example_1]
}
```